source/common/http/conn_manager_impl.cc

```diff
@@ -366,6 +367,7 @@ RequestDecoder& ConnectionManagerImpl::newStream(ResponseEncoder& response_encod
   new_stream->response_encoder_->getStream().addCallbacks(*new_stream);
   new_stream->response_encoder_->getStream().setFlushTimeout(new_stream->idle_timeout_ms_);
   new_stream->streamInfo().setDownstreamBytesMeter(response_encoder.getStream().bytesMeter());
+  new_stream->streamInfo().setStreamState(StreamInfo::StreamState::Started);
```

Comment:
I can't think of a case when the intermediate state before Started is useful. Is there some use case when we create a stream but it is not started?

Reply:
As per my other comment, IMO the Started state should mark an upstream connection or some kind of engagement. If it's not set yet, it means it didn't happen yet. Therefore we can distinguish between - not contacting upstream yet, upstream connection was made, upstream stream is active and upstream stream has ended.